app/models/miq_group.rb

```diff
@@ -297,10 +297,11 @@ def current_user_group?
   end
 
   def ensure_can_be_destroyed
-    raise _("The login group cannot be deleted") if current_user_group?
-    raise _("The group has users assigned that do not belong to any other group") if single_group_users?
-    raise _("A tenant default group can not be deleted") if tenant_group? && referenced_by_tenant?
-    raise _("A read only group cannot be deleted.") if system_group?
+    errors.add(:base, _("The login group cannot be deleted")) if current_user_group?
+    errors.add(:base, _("The group has users assigned that do not belong to any other group")) if single_group_users?
+    errors.add(:base, _("A tenant default group can not be deleted")) if tenant_group? && referenced_by_tenant?
+    errors.add(:base, _("A read only group cannot be deleted.")) if system_group?
+    throw :abort unless errors[:base].empty?
```

Comment:
But here you check if `errors[:base].empty?` Couldn't it have have stale content  if you don't clear it?

Reply:
`errors` belongs to instance and for each record user attempting to delete new empty `errors` will be available

Reply:
I think it is extra to clear `errors` inside `before_destroy`

Reply:
I can see @jvlcek's point. But I'm also thinking that clearing errors here would be wrong because it could end up clearing errors that were added from a different validation that may have failed but ran in the same validation chain